pkg/plugin/v1/init.go

```diff
@@ -0,0 +1,169 @@
+
+package v1
+
+import (
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/spf13/pflag"
+	"sigs.k8s.io/kubebuilder/pkg/model/config"
+	"sigs.k8s.io/kubebuilder/pkg/plugin"
+	"sigs.k8s.io/kubebuilder/pkg/plugin/scaffold"
+
+	"github.com/joelanford/helm-operator/pkg/internal/kubebuilder/cmdutil"
+	"github.com/joelanford/helm-operator/pkg/internal/kubebuilder/validation"
+	"github.com/joelanford/helm-operator/pkg/plugin/internal/chartutil"
+	"github.com/joelanford/helm-operator/pkg/plugin/v1/scaffolds"
+)
+
+type initPlugin struct {
+	config        *config.Config
+	apiPlugin     createAPIPlugin
+	doAPIScaffold bool
+
+	// For help text.
+	commandName string
+}
+
+var (
+	_ plugin.Init        = &initPlugin{}
+	_ cmdutil.RunOptions = &initPlugin{}
+)
+
+func (p *initPlugin) UpdateContext(ctx *plugin.Context) {
+	ctx.Description = `Initialize a new Helm-based operator project.
+
+Writes the following files:
+- a helm-charts directory with the chart(s) to build releases from
+- a watches.yaml file that defines the mapping between your API and a Helm chart
+- a PROJECT file with the domain and repo
+- a Makefile to build the project
+- a Kustomization.yaml for customizating manifests
+- a Patch file for customizing image for manager manifests
+- a Patch file for enabling prometheus metrics
+`
+	ctx.Examples = fmt.Sprintf(`  $ %s init --plugins=%s \
+      --domain=example.com \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --group=apps --version=v1alpha1 \
+      --kind=AppService \
+      --helm-chart=myrepo/app
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=myrepo/app
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=myrepo/app \
+      --helm-chart-version=1.2.3
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=app \
+      --helm-chart-repo=https://charts.mycompany.com/ \
+      --helm-chart-version=1.2.3
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=/path/to/local/chart-directories/app/
+
+  $ %s init --plugins=%s \
+      --domain=example.com \
+      --helm-chart=/path/to/local/chart-archives/app-1.2.3.tgz
+`,
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+		ctx.CommandName, plugin.KeyFor(Plugin{}),
+	)
+
+	p.commandName = ctx.CommandName
+}
+
+func (p *initPlugin) BindFlags(fs *pflag.FlagSet) {
+	fs.SortFlags = false
+	fs.StringVar(&p.config.Domain, "domain", "my.domain", "domain for groups")
+	p.apiPlugin.BindFlags(fs)
```

Comment:
missing the repo flag. We need to get or gen the repo to add in the PROJECT file. It is used by SDK features (bundle)

here: 

```go

	fs.StringVar(&p.config.Repo, "repo", "", "name to use for go module (e.g., github.com/user/repo), "+
		"defaults to the go package of the current working directory.")
```

In the validate: 
```go
// Try to guess repository if flag is not set.
	if p.config.Repo == "" {
		repoPath, err := util.FindCurrentRepo()
		if err != nil {
			return fmt.Errorf("error finding current repository: %v", err)
		}
		p.config.Repo = repoPath
	}

``` 

 util.FindCurrentRepo()->  cames from https://github.com/kubernetes-sigs/kubebuilder/blob/master/pkg/plugin/internal/util/repository.go